lessons/intro/pathlib/index.md

```diff
@@ -0,0 +1,621 @@
+Je to normální soubor? True
+```
+
+Objekty `Path` lze používat v naprosté většině situací, kdy jde použít cesta
+jako řetězec.
+Například pro funkci `open`:
+
+```python
+with open(ententyky, encoding='utf-8') as file:
+    print(file.read())
+```
+
+`Path` ale má `open` i jako metodu:
+
+```python
+with ententyky.open(encoding='utf-8') as file:
+    print(file.read())
+```
+
+A protože je čtení celého textového obsahu souboru docela užitečné,
+existuje i zkratka která soubor otevře, přečte a zavře najednou:
+
+```python
+print(ententyky.read_text())
+```
+
+(Větší soubory je ale lepší otevřít ve `with` a zpracovávat třeba po řádcích,
+aby se obsah nemusel do paměti počítače načíst celý najednou.)
+
+Existuje i `write_text`:
+
+```python
+cesta = Path.cwd() / 'pisnicka.txt'
+cesta.write_text('Holka modrooká\nNesedávej u potoka!')
+```
+
+{#
+## Zpátky na stromy... tedy řetězce
+
+Ve většině případů jde Path použít tam, kde se cesta dá zadat jako řetězec. Na ale vždy – pathlib existuje teprve od roku 2014, a některé Pythonní knihovny stále ještě vyžadují řetězce.
+
+Jedna z výjimek je IPython.display.Image, která umí v Notebooku vykreslit obrázek.
+
+```pycon
+>>> from IPython.display import Image
+```
+
+Image potřebuje (aspoň začátkem roku 2018) řetězcovou cestu.
+Příkaz `Image(base / 'hrad.jpeg')` mi skončil chybou typu –
+`TypeError: a bytes-like object is required, not 'PosixPath'`.
+
+V takových případech ale stačí Path převést na řetězec.
+
+```pycon
+>>> str(base / 'hrad.jpeg')
+'soubory/hrad.jpeg'
+
+>>> Image(str(base / 'hrad.jpeg'))
+
+>>> # Informace o autorství obrázku (díky, Millenium187!)
+>>> print(base.joinpath('hrad.attribution').read_text())
+
+"hrad.jpg" is (c) 2011, Wikipedia user Millenium187
+    https://commons.wikimedia.org/wiki/User:Millenium187
+Here used under the Creative Commons Attribution-Share Alike 3.0 Unported license.
+
+See: https://commons.wikimedia.org/wiki/File:Hrad_%C5%A0pilberk_II.jpg
+```
+#}
+
+## A co adresáře?
+
+I s adresáři umí `pathlib` pracovat.
+Nejzákladnější operace je získání cest k obsaženým souborům:
+
+```pycon
+>>> zaklad.iterdir()
+<generator object Path.iterdir at 0x7fbd4443b9e8>
+```
+
+Metoda iterdir opět vrací *iterátor* – objekt, přes který musíš „projít“
+(cyklem for, převedením na seznam ap.), abys z něj dostal{{a}} obsah.
+
+```pycon
+>>> list(zaklad.iterdir())
+[PosixPath('soubory/hrad.jpeg'),
+ PosixPath('soubory/hrad.attribution'),
+ PosixPath('soubory/.gitignore'),
+ PosixPath('soubory/kolecko.png'),
+ PosixPath('soubory/texty'),
+ PosixPath('soubory/hlad.txt'),
+ PosixPath('soubory/hraz.attribution'),
+ PosixPath('soubory/ententyky.txt'),
+ PosixPath('soubory/hraz.jpeg'),
+ PosixPath('soubory/README')]
+
+>>> for cesta in zaklad.iterdir():
+>>>    print(cesta)
+soubory/hrad.jpeg
+soubory/hrad.attribution
+soubory/.gitignore
+soubory/kolecko.png
+soubory/texty
+soubory/hlad.txt
+soubory/hraz.attribution
+soubory/ententyky.txt
+soubory/hraz.jpeg
+soubory/README
+```
+
+{#
```

Comment:
Comment?

Reply:
Tady si taky nejsem jistý, jestli je to dobré probírat. Vysvětlit globy zabere dost času.

Reply:
To je pravda, ale když už je to jednou napsáno tak bych to tam nechal. Navíc je to na konci, takže ideální kandidát na bonusové samostudium.

Případně je možné z toho udělat samostatné lekce.